File: apps/api/app/schemas/delivery.py
from datetime import datetime

from pydantic import BaseModel, Field


class DeliveryPostCreate(BaseModel):
    title: str
    content: str
    foodCategory: str
    orderLink: str | None = None
    bankAccount: str | None = None
    maxParticipants: int = Field(..., ge=1)


class DeliveryAuthor(BaseModel):
    id: str
    nickname: str | None


class DeliveryParticipantInfo(BaseModel):
    id: str
    nickname: str | None


class DeliveryPostResponse(BaseModel):
    id: str
    title: str
    content: str
    foodCategory: str
    orderLink: str | None
    bankAccount: str | None
    maxParticipants: int
    isClosed: bool
    createdAt: datetime
    author: DeliveryAuthor
    participants: list[DeliveryParticipantInfo]


class DeliveryPostListItem(BaseModel):
    id: str
    title: str
    foodCategory: str
    maxParticipants: int
    isClosed: bool
    createdAt: datetime
    author: DeliveryAuthor
    participantCount: int


class DeliveryPostListResponse(BaseModel):
    data: list[DeliveryPostListItem]

File: apps/mobile/src/types/delivery.ts
export type FoodCategory =
  | 'KOREAN'
  | 'CHINESE'
  | 'JAPANESE'
  | 'WESTERN'
  | 'FASTFOOD'
  | 'CHICKEN'
  | 'PIZZA'
  | 'DESSERT'
  | 'OTHER';

export interface DeliveryAuthor {
  id: string;
  nickname: string | null;
}

export interface DeliveryParticipant {
  id: string;
  nickname: string | null;
}

export interface DeliveryPostListItem {
  id: string;
  title: string;
  foodCategory: string;
  maxParticipants: number;
  isClosed: boolean;
  createdAt: string;
  author: DeliveryAuthor;
  participantCount: number;
}

export interface DeliveryPost {
  id: string;
  title: string;
  content: string;
  foodCategory: string;
  orderLink: string | null;
  bankAccount: string | null;
  maxParticipants: number;
  isClosed: boolean;
  createdAt: string;
  author: DeliveryAuthor;
  participants: DeliveryParticipant[];
}

export interface DeliveryPostListResponse {
  data: DeliveryPostListItem[];
}

export interface CreateDeliveryPostRequest {
  title: string;
  content: string;
  foodCategory: string;
  orderLink?: string;
  bankAccount?: string;
  maxParticipants: number;
}

export interface DeliveryMessage {
  id: string;
  senderId: string;
  senderNickname: string | null;
  content: string;
  createdAt: string;
}

export interface DeliverySystemMessage {
  message: string;
  userId?: string;
}

export type DeliveryWebSocketMessage =
  | { type: 'message'; data: DeliveryMessage }
  | { type: 'system'; data: DeliverySystemMessage };
